app/models/proxy_rule.rb

```diff
@@ -25,6 +28,16 @@ class ProxyRule < ApplicationRecord
 
   ALLOWED_HTTP_METHODS = %w[GET POST DELETE PUT PATCH HEAD OPTIONS].freeze
 
+  scope :by_query, ->(query, owner_type, owner_id) do
+    return if query.blank?
```

Comment:
```suggestion
    return none if query.blank?
```
Not sure if it will work, but then this way we return the same time and we can chain scopes... so it won't be null anymore. Btw, lambda checks that the params are passed so 'query' should never be nil :thinking: so imo we shouldn't even have this checking the type at all

Reply:
My first shoot was returning a `none` relation. But it doesn't work because it ignores prepended and postponed relations in the query. eg:

with `return none ..` 
```
[1] pry(main)> ProxyRule.by_query('', 'Proxy', 22).where(position: 2).to_sql
=> ""
```
it returned the empty relation and ignored the where position.

with `return nil`
```
[1] pry(main)> ProxyRule.by_query('', 'Proxy', 22).where(position: 2).to_sql
=> "SELECT `proxy_rules`.* FROM `proxy_rules` WHERE `proxy_rules`.`position` = 2"
```
it skipped the where(id: ..) and made the `where(position: )` one.

Reply:
This could be a query see `app/queries`